stan-native/src/main/java/org/ballerinalang/nats/streaming/consumer/StreamingListener.java

```diff
@@ -116,11 +118,19 @@ private void executeResource(String subject, Object[] args, Type returnType) {
                                                                           connectedUrl, subject);
             properties.put(ObservabilityConstants.KEY_OBSERVER_CONTEXT, observerContext);
             runtime.invokeMethodAsync(service, ON_MESSAGE_RESOURCE,
-                                      null, metadata, new DispatcherCallback(connectedUrl, subject),
-                                      properties, returnType, args);
+                    null, metadata,
+                    new DispatcherCallback(connectedUrl, subject, countDownLatch),
+                    properties, returnType, args);
         } else {
             runtime.invokeMethodAsync(service, ON_MESSAGE_RESOURCE, null, metadata,
-                                      new DispatcherCallback(connectedUrl, subject), args);
+                                      new DispatcherCallback(connectedUrl, subject, countDownLatch), args);
+        }
+        try {
+            countDownLatch.await();
```

Comment:
Instead of `CountDownLatch` shall we use `environment.markAsync()` and mark it complete at the callback?

That way we can make this non-blocking. (May have to pass the environment here though).

Reply:
@ThisaruGuruge tried and there is a issue :/

Reply:
Shall we merge this like this for now? WDYT

Reply:
+1